### PR TITLE
Service Layer Refactoring

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- removed the service layer
- routed the request directly to the controller

### CODE REVIEW

File: controller/encounter_controller.py [hunk 1]
```diff
@@ -75,6 +75,24 @@ def update_by_patient_id(patient_id: str , enc: EncounterUpdateModel):
             return Response(
                 content=f"Error: Unable to update encounter", status_code=status.HTTP_400_BAD_REQUEST
             )
+    
+    @staticmethod
+    def get_all_encounters():
+        try:
+            encounters = API.do_request(method = "GET", endpoint= f"/fhir/Encounter")
```
Comment (tamsud) [inline controller/encounter_controller.py:82]: try get from base

File: controller/hl7_immunization_controller.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            logger.error(f"Unable to get immunization data: {str(e)}")
+            logger.error(traceback.format_exc())
+            return Response(
+                content=f"Error: No Immunization data found for the {patient_id}", status_code=status.HTTP_400_BAD_REQUEST
+            )
+        
+    @staticmethod
+    def get_all_immunizations():
+        try:
+            response_immunization = API.do_request(method = "GET", endpoint= "/fhir/Immunization")
```
Comment (tamsud) [inline controller/hl7_immunization_controller.py:53]: update the aidbox package to the latest version.
And use the make_request instead of do_request

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/patient_controller.py [hunk 3]
```diff
@@ -87,6 +87,47 @@ def get_all_patients():
                 content=f"Error: Unable to retrie patients",
                 status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
             )
+        
+    @staticmethod
+    def update_patient_by_id(patient_id , pat: PatientUpdateModel):
+        try:
```
Comment (tamsud) [inline controller/patient_controller.py:93]: Is that all parameters are mandatory?

File: routes/medication_routes.py [hunk 4]
```diff
@@ -1,27 +1,22 @@
...
+@router.post("/medication", response_model=dict)
+async def create_medication(med: MedicationCreateModel):
+    return MedicationClient.create_medication(med)
 
 
 @router.get("/medication/{patient_id}")
-@permission_required("MEDICATION", "READ")
-def get_medication_by_patient_id(patient_id: str, request: Request):
-    return MedicationService.get_medication_by_patient_id(patient_id)
+def get_medication_by_patient_id(patient_id: str):
```
Comment (tamsud) [inline routes/medication_routes.py:15]: In router change all the API calls to the async fucntions

File: routes/medication_routes.py [hunk 5]
```diff
@@ -1,27 +1,22 @@
...
+    return MedicationClient.get_medication_by_patient_id(patient_id)
 
 
 @router.put("/medication/{patient_id}")
-@permission_required("MEDICATION", "UPDATE")
-async def update_medication(patient_id: str, updated_medication: MedicationUpdateModel, request: Request):
-    return MedicationService.update_medication_by_patient_id(patient_id, updated_medication)
+async def update_medication(patient_id: str, updated_medication: MedicationUpdateModel):
+    return MedicationClient.update_medication_by_patient_id(patient_id, updated_medication)
```
Comment (tamsud) [inline routes/medication_routes.py:21]: image
image
fucntion is expecting only one argument why two arguments are passed into it

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/encounter_controller.py (hunk 1 above)
Comment (MonishaSivanathan) [reply controller/encounter_controller.py]: i have got from base and changed the code

File: controller/hl7_immunization_controller.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                raise Exception("Unable to create the record")
+        except Exception as e:
+            logger.error(f"Unable to create a Immunization: {str(e)}")
+            logger.error(traceback.format_exc())
+            return Response(content=response_data, status_code=status.HTTP_400_BAD_REQUEST)
+        
+    @staticmethod
+    def get_immunizations_by_patient_id(patient_id: str):
+        try:
+            response_immunization = API.do_request(method = "GET", endpoint= f"/fhir/Immunization/?patient=Patient/{patient_id}")
```
[the remark this answers is not in the document]
Comment (MonishaSivanathan) [reply controller/hl7_immunization_controller.py]: i have got from base and changed the do_request to make_request

File: controller/patient_controller.py (hunk 3 above)
Comment (MonishaSivanathan) [reply controller/patient_controller.py:93]: it is not i have removed it.

File: routes/medication_routes.py (hunk 4 above)
Comment (MonishaSivanathan) [reply routes/medication_routes.py]: i have changed to async function

File: routes/medication_routes.py (hunk 5 above)
Comment (MonishaSivanathan) [reply routes/medication_routes.py]: sorry, i have resolved it.

**Review by tamsud — APPROVED** [review]

**Pull request merged**